server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -3305,21 +3305,24 @@ public void testSyncConcurrently() throws Exception {
                             }
                             assertNotNull(location);
                             long globalCheckpoint = lastGlobalCheckpoint.get();
+                            final boolean synced;
                             if (randomBoolean()) {
-                                translog.ensureSynced(location);
+                                synced = translog.ensureSynced(location);
                             } else {
-                                translog.sync();
+                                synced = translog.sync();
```

Comment:
I think we can simply set synced=true here in order to always do the check later. Since `sync()` always sync's if GCP is different from last sync'ed GCP, this should be OK. That also avoids the signature change on `Translog.sync()`